x-pack/filebeat/input/entityanalytics/provider/okta/internal/okta/ratelimiter.go

```diff
@@ -16,33 +16,66 @@ import (
 	"github.com/elastic/elastic-agent-libs/logp"
 )
 
-type RateLimiter map[string]*rate.Limiter
+type RateLimiter struct {
+	window     time.Duration
+	fixedLimit *int
+	byEndpoint map[string]endpointRateLimiter
+}
+
+type endpointRateLimiter struct {
+	limiter *rate.Limiter
+	ready   chan struct{}
+}
 
-func NewRateLimiter() RateLimiter {
-	r := make(RateLimiter)
-	return r
+const waitDeadline = 30 * time.Minute
```

Comment:
Deadline conventionally is a `time.Time`. `waitTimeout`? I think this wants godoc too, to explain what its intention is.